analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/ValidatorTestCheck.cs

```diff
@@ -51,6 +52,12 @@ internal class ValidatorTestCheck : SymbolicCheck
         public override void ExecutionCompleted() =>
             executionCompletedCount++;
 
+        public ISymbol Symbol(string name)
+        {
+            symbols.Should().ContainKey(name, "asserted symbols should be in the compilation. Missing symbol: " + name);
+            return symbols[name];
+        }
```

Comment:
Can be a single line:
```suggestion
        public ISymbol Symbol(string name) =>
            symbols.Should().ContainKey(name, "asserted symbols should be in the compilation").WhichValue;
```
The "Missing symbol:" part is superfluous. The key is (most likely) called out in the assertion message itself already. If you want to keep it, use the integrated formatting support: `"asserted symbols should be in the compilation. Missing symbol: {0}", name)`.

Reply:
I'd prefer to keep the assertion away from the action.
I've updated the formatting, it makes the message nicer.